src/main/java/io/syndesis/verifier/v1/metadata/MetadataAdapter.java

```diff
@@ -0,0 +1,12 @@
+package io.syndesis.verifier.v1.metadata;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+import org.apache.camel.component.extension.MetaDataExtension.MetaData;
+
+@FunctionalInterface
+public interface MetadataAdapter extends BiFunction<Map<String, Object>, MetaData, Map<String, List<PropertyPair>>> {
```

Comment:
a short explanation what the adapter should adapt to which format would be awesome here (i.e. what are the key/values of those maps)